tests/e2e/common/diagnostics/options.go

```diff
@@ -0,0 +1,97 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package diagnostics
+
+import (
+	"k8s.io/client-go/kubernetes"
+	nfdclient "sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned"
+)
+
+type Diagnostic struct {
+	*Config
+	collectors []Collector
+}
+
+type Option func(*Diagnostic)
+
+func WithNamespace(namespace string) func(*Diagnostic) {
+	return func(d *Diagnostic) {
+		d.Config.namespace = namespace
+	}
+}
+
+func WithArtifactDir(artifactDir string) func(*Diagnostic) {
+	return func(d *Diagnostic) {
+		d.Config.artifactDir = artifactDir
+	}
+}
+
+func WithKubernetesClient(clientset *kubernetes.Clientset) func(*Diagnostic) {
+	return func(d *Diagnostic) {
+		d.clientset = clientset
+	}
+}
+
+func WithNFDClient(nfdClient *nfdclient.Clientset) func(*Diagnostic) {
+	return func(d *Diagnostic) {
+		d.NfdClient = nfdClient
+	}
+}
+
+func WithObjects(objects ...string) func(*Diagnostic) {
+	return func(d *Diagnostic) {
+		seen := make(map[string]bool)
+		for _, obj := range objects {
+			if seen[obj] {
+				continue
+			}
+			seen[obj] = true
+			switch obj {
```

Comment:
Nit: We don't handle the case where an input is provided that's not a supported object.